osquery/tables/events/windows/ntfs_journal_events.cpp

```diff
@@ -0,0 +1,373 @@
+
+  for (const auto& event : ec->event_list) {
+    if (!shouldEmit(sc, event)) {
+      continue;
+    }
+
+    auto row = generateRowFromEvent(event);
+    row["category"] = TEXT(sc->category);
+    emitted_row_list.push_back(row);
+  }
+
+  addBatch(emitted_row_list);
+
+  return Status(0);
+}
+
+void processConfiguration(const NTFSEventSubscriptionContextRef context,
+                          const StringList& access_categories,
+                          StringList& include_paths,
+                          StringList& exclude_paths) {
+  // clang-format off
+  auto path_erase_it = std::remove_if(
+    include_paths.begin(),
+    include_paths.end(),
+
+    [exclude_paths](const std::string &str) -> bool {
+      auto it = std::find(exclude_paths.begin(), exclude_paths.end(), str);
+      return it != exclude_paths.end();
+    }
+  );
+  // clang-format on
+
+  if (path_erase_it != include_paths.end()) {
+    include_paths.erase(path_erase_it, include_paths.end());
+  }
+
+  if (include_paths.empty()) {
+    return;
+  }
+
+  std::unordered_set<USNFileReferenceNumber> frn_set;
+
+  // Build the FRN set from the now-filtered paths.
+  // NOTE(ww): path can be either a file or a directory,
+  // so we need to pass FILE_FLAG_BACKUP_SEMANTICS rather
+  // than FILE_ATTRIBUTE_NORMAL.
+  for (const auto& path : include_paths) {
+    HANDLE file_hnd = ::CreateFile(path.c_str(),
+                                   GENERIC_READ,
+                                   FILE_SHARE_READ | FILE_SHARE_WRITE,
+                                   NULL,
+                                   OPEN_EXISTING,
+                                   FILE_FLAG_BACKUP_SEMANTICS,
+                                   NULL);
+
+    // resolveFilePattern should filter out any nonexistent files,
+    // so this should never be the case (except for TOCTOU).
+    if (file_hnd == INVALID_HANDLE_VALUE) {
+      TLOG << "Couldn't open " << path << " while buiding FRN set";
+      continue;
+    }
+
+    // NOTE(ww): This shouldn't fail once we have a valid handle, but there's
+    // another TOCTOU here: another process could delete the file before we
+    // get its information. We don't want to lock the file, though, since it
+    // could be something imporant used by another process.
+    USNFileReferenceNumber frn = 0;
+
+#if _WIN32_WINNT > _WIN32_WINNT_WIN7
+    FILE_ID_INFO file_id_info;
+    if (!::GetFileInformationByHandleEx(
+            file_hnd, FileIdInfo, &file_id_info, sizeof(file_id_info))) {
+      TLOG << "Couldn't get FRN for " << path << " while building FRN set";
+      ::CloseHandle(file_hnd);
+      continue;
+    }
+
+    frn = file_id_info.FileId;
+#else
+    BY_HANDLE_FILE_INFORMATION file_id_info;
+    if (!::GetFileInformationByHandle(file_hnd, &file_id_info)) {
+      TLOG << "Couldn't get FRN for " << path << " while building FRN set";
+      ::CloseHandle(file_hnd);
+      continue;
+    }
+
+    frn = ((long long)file_id_info.nFileIndexHigh << 32) |
```

Comment:
Should we static_cast this to long long?

Reply:
Yep. I'll also make it `DWORDLONG` for consistency with `operator=`.